### PR TITLE
Use of isEmpty() and improve javadoc

### PR DESCRIPTION
**Pull request by ChristosToxiadis**
I guess that the parameter delimiter was meant to be @NonNull so i changed the if condition.

### CODE REVIEW

File: src/main/java/org/plumelib/util/StringsPlume.java [hunk 1]
```diff
@@ -308,6 +308,7 @@ public static String escapeJava(String orig) {
       char c = orig.charAt(i);
       switch (c) {
         case '\"':
+        case '\'':
```
Comment (mernst) [inline src/main/java/org/plumelib/util/StringsPlume.java:311]: This isn't needed for strings.

Also, please add a test when you change behavior.

File: src/main/java/org/plumelib/util/StringsPlume.java [hunk 2]
```diff
@@ -372,6 +373,8 @@ public static String escapeJava(char c) {
     switch (c) {
       case '\"':
         return "\\\"";
+      case '\'':
+        return "\\'";
```
Comment (mernst) [inline src/main/java/org/plumelib/util/StringsPlume.java:377]: This is not needed either.

File: src/main/java/org/plumelib/util/StringsPlume.java [hunk 3]
```diff
@@ -422,6 +425,8 @@ public static String escapeNonASCII(String orig) {
   private static String escapeNonASCII(char c) {
     if (c == '"') {
       return "\\\"";
+    } else if (c == '\''){
```
Comment (mernst) [inline src/main/java/org/plumelib/util/StringsPlume.java:428]: Please use spacing that is consistent with the surrounding code: add a space within "){".  (I fixed this.)

File: src/main/java/org/plumelib/util/StringsPlume.java [hunk 4]
```diff
@@ -600,10 +605,11 @@ public static String removeWhitespaceAround(String arg, String delimiter) {
    * @param arg string to remove whitespace in
    * @param delimiter string to remove whitespace after
    * @return version of arg, with whitespace after delimiter removed
+   * @throws IllegalArgumentException if the second argument is an empty String
```
Comment (mernst) [inline src/main/java/org/plumelib/util/StringsPlume.java:608]: Adding `@throws`  to the specification is usually the wrong thing to do.  It is better to state the requirements, and not specify what happens if they are violated.  You don't want clients to depend on the specific exception, and you don't want to restrict the flexibility to change the implementation in the future.

Here you are right that the specification is incomplete and should be improved -- thanks for noticing that.  However, a better fix is to change
```java
   * @param delimiter string to remove whitespace after
```
to
```java
   * @param delimiter a non-empty string to remove whitespace after
```

File: src/main/java/org/plumelib/util/StringsPlume.java [hunk 5]
```diff
@@ -634,10 +640,11 @@ public static String removeWhitespaceAfter(String arg, String delimiter) {
    * @param arg string to remove whitespace in
    * @param delimiter string to remove whitespace before
    * @return version of arg, with whitespace before delimiter removed
+   * @throws IllegalArgumentException if the second argument is an empty String
```
Comment (mernst) [inline src/main/java/org/plumelib/util/StringsPlume.java:643]: Similar comment as above.

**Review by mernst — CHANGES_REQUESTED** [review]
This pull request changes the contract of `escapeJava` (it becomes useful for both string quoting and character quoting, and in each context it adds some backslashes that are not necessary), without changing the documentation or tests.

It is confusing that `escapeJava(char)` does quoting for a string literal (not for a char literal), so in [a separate pull request](https://github.com/plume-lib/plume-util/pull/125) I have clarified its documentation and also deprecated it.  That obviates some but not all of the changes in this pull request.

File: src/main/java/org/plumelib/util/StringsPlume.java (hunk 1 above)
Comment (ChristosToxiadis) [reply src/main/java/org/plumelib/util/StringsPlume.java:311]: Sorry about that, it was very stupid. I don't know what i was thinking.

File: src/main/java/org/plumelib/util/StringsPlume.java (hunk 4 above)
Comment (ChristosToxiadis) [reply src/main/java/org/plumelib/util/StringsPlume.java:608]: I added @thows because i saw it a lot in documentation of other methods in this repo.

Comment (mernst) [reply src/main/java/org/plumelib/util/StringsPlume.java:608]: In this file, there are only 5 `@throws`, and 4 were just added by you.  So I think there are not a lot of `@throws` in this file before your changes.

**Comment by mernst** [comment]
@ChristosToxiadis There are a few good changes in this pull request.  You could prune it down to them, or I could create a new pull request that includes just those and close this one.

**Comment by ChristosToxiadis** [comment]
I could create a new pull request based on your review. So, i have to delete the changes in the escape methods, except escapeNonASCII, to change the documentation and to fix the spacing. If i am missing something, tell me so i can include it.

**Comment by mernst** [comment]
I agree that it might be easiest to create a new pull request.  You can look at the current code and see what, if anything, needs to be improved.  Thanks!

File: src/main/java/org/plumelib/util/StringsPlume.java (hunk 4 above)
Comment (ChristosToxiadis) [reply src/main/java/org/plumelib/util/StringsPlume.java:687]: I was referring to the whole repository. For example in Arrays Plume there are more than 15 @throws.

https://github.com/plume-lib/plume-util/blob/master/src/main/java/org/plumelib/util/ArraysPlume.java#L375

**Comment by ChristosToxiadis** [comment]
Ok i will close this one then.

**Pull request closed without merging**